distributed/worker.py

```diff
@@ -1893,6 +1894,7 @@ async def plugin_remove(self, name: str) -> ErrorMessage | OKMessage:
                 if isawaitable(result):
                     result = await result
         except Exception as e:
+            logger.exception("Worker plugin %s failed to teardown", name)
```

Comment:
Might be worth adding a corresponding test for this as well. I don't mean this as a blocking comment though

Reply:
Currently working on this (committed this too early)

Reply:
Done